### DISCUSSION TITLE
Insert a value at cursor's position

### DISCUSSION THREAD

**Discussion by ungureanustefan** [mdx-editor/editor#147] [Q&A]
Hi,

I've tried to extend the library and use Lexical to create a plugin for a dropdown button with a list of strings such as "[{name: "Name", value: "name}]". When the user clicks in the middle of the text and then selects an item from the dropdown, the value should be inserted where the cursor is.

I'm really stuck and I have no idea how to do it, because everything I've tried so far was a dead end. I've tried to look at the source code and also to access Lexical's state, but no success. So if anyone has a bit spare time, could you please show me what should be done in order to achieve that functionality?

Thank you.

**Comment by petyosi** [comment]
Can you show what you've done so far?

**Reply by ungureanustefan** [reply]
@petyosi,
Most of the stuff that I've tried were a dead end and I've removed them. The last trial, however, that I attempted, was to access the "editor" as this guy does in the video here [https://www.youtube.com/watch?v=qIqxvk2qcmo](https://www.youtube.com/watch?v=qIqxvk2qcmo). But by doing that, it breaks the app. I'm not sure how to extend the library in React to achieve that functionality.

[This is the code at the moment](https://codesandbox.io/p/sandbox/stoic-napier-232h24?file=%2Fsrc%2FApp.tsx%3A34%2C53)

I'm not sure where to go from here. Lexical is installed in node_modules because it is a dependency of mdxeditor, but I cannot access its state for whatever reason.

**Reply by petyosi** [reply]
The best way for you to move forward is to read the source code of some of the plugins and the built in toolbar tools, and the way they interact with the core plugin state management. For example, this is [the implementation of inserting a decorator node](https://github.com/mdx-editor/editor/blob/main/src/plugins/core/index.ts#L397-L427). 

You can use one of the simpler plugins (like the headings one) as a starting point.

**Reply by ungureanustefan** [reply]
@petyosi Should I fork the library into a github repo and extend it in that way, or there's a way to extend it straight from my React App?

**Reply by petyosi** [reply]
You don't need to fork the library. Everything necessary to create a custom plugin is publicly exported.

**Comment by petyosi** [comment]
Sometime after this discussion, the editor implemented a method `insertMarkdown` that will inject a given markdown at the cursor position.
